warcprox/mitmproxy.py

```diff
@@ -416,14 +426,15 @@ def _proxy_request(self, extra_response_headers={}):
             req += self.rfile.read(int(self.headers['Content-Length']))
 
         prox_rec_res = None
+        connection_is_fine = False
```

Comment:
Minor point, but I'd suggest reorganizing the code as follows to avoid needing this `connection_is_fine` variable:
```
        try:
            ...
            if not is_connection_dropped(self._remote_server_conn):
                self._conn_pool._put_conn(self._remote_server_conn)
            else:
                self._remote_server_conn.sock.close()
        except:
            self._remote_server_conn.sock.close()
        finally:
            if prox_rec_res:
                prox_rec_res.close()
```

Reply:
This is excellent, I did that.